jsdom-nodejs-env/src/main/scala/org/scalajs/jsenv/jsdomnodejs/JSDOMNodeJSEnv.scala

```diff
@@ -42,24 +43,31 @@ class JSDOMNodeJSEnv(config: JSDOMNodeJSEnv.Config) extends JSEnv {
   def startWithCom(input: Input, runConfig: RunConfig,
       onMessage: String => Unit): JSComRun = {
     JSDOMNodeJSEnv.validator.validate(runConfig)
+    validateInput(input)
     ComRun.start(runConfig, onMessage) { comLoader =>
-      val files = initFiles ::: (comLoader :: codeWithJSDOMContext(input))
-      internalStart(files, runConfig)
+      internalStart(comLoader :: Nil, input, runConfig)
     }
   }
 
-  private def internalStart(files: List[VirtualBinaryFile],
+  private def validateInput(input: Input): Unit = {
+    input match {
+      case _:Input.ScriptsToLoad =>
+        // ok
+      case _ =>
+        throw new UnsupportedInputException(input)
+    }
+  }
```

Comment:
Can we consolidate this with `scriptFiles` so that we do not have the logic twice?

Reply:
I've pushed another commit where `validateInput` calls `scriptFiles`. That removes the duplication, but it does "more work" than necessary in `validateInput`. Let me know which one you prefer and I'll either squash or revert the commit.